python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -495,9 +496,27 @@ def _get_state_dict(self, state):
         }
 
     def _update_stats_metrics(self, state: str, force_update: bool = False):
+        # include budgets
+        metrics = []
+        for op in self._topology:
+            budget = self._resource_manager.get_budget(op)
+            if budget is not None:
+                # Convert inf to -1 to represent unlimited budget in metrics
+                op.metrics.cpu_budget = -1 if math.isinf(budget.cpu) else budget.cpu
+                op.metrics.gpu_budget = -1 if math.isinf(budget.gpu) else budget.gpu
+                op.metrics.memory_budget = (
+                    -1 if math.isinf(budget.memory) else budget.memory
+                )
+                op.metrics.object_store_memory_budget = (
+                    -1
+                    if math.isinf(budget.object_store_memory)
+                    else budget.object_store_memory
+                )
+                metrics.append(op.metrics)
+
```

Comment:
My initial reaction is that this breaks encapsulation. Like, I feel like the abstractions aren't great if `StreamingExecutor` is directly setting attributes on `OpRuntimeMetrics` and extending `op.metrics`.

Would this be cleaner if we directly defined new gauges in `ResourceManager` and update those there?